lib/game_stats.rb

```diff
@@ -1,10 +1,116 @@
-class GameStats
-  
-  def initialize(data)
-    @games = data.games
-    @teams = data.teams
-    @game_teams = data.game_teams
-  end
-  
-  
-end
+require "csv"
+require "stat_tracker"
+require "pry"
+
+class GameStats
+  def initialize(data)
+    @games = data.games
+    @teams = data.teams
+    @game_teams = data.game_teams
+  end
+
+  def highest_total_score
+    all_goals.max
+  end
+
+  def lowest_total_score
+    all_goals.min
+  end
+
+  def percentage_home_wins
+    home_wins = []
+    all_home_games = []
+
+    @game_teams.each do |row|
+      home_wins << row if row[:hoa] == "home" && row[:result] == "WIN"
+      all_home_games << row if row[:hoa] == "home"
+    end
+
+    ((home_wins.count / all_home_games.count.to_f).round(2))
+  end
+
```

Comment:
You could combine percentage_home_wins, percentage_visitor_wins, and percentage_ties into one helper method: 
`def percentage_result(num_condition, denom_condition)`
  `  num_count = 0`
  `  denom_count = 0`
  `  @game_teams.each do |row|`
    `    num_count += 1 if num_condition`
    `    denom_count += 1 if denom_condition`
  `  end`
  `  (num_count / denom_count.to_f).round(2)`
`end`

For example, the `percentage_home_wins` method could look like this:
`def percentage_home_wins`
  `  home_wins = row[:hoa] == "home" && row[:result] == "WIN"`
  `  all_home_games = row[:hoa] == "home"`
  `  percentage_result(home_wins, all_home_games)`
`end`

Reply:
@neb417 This could potentially be a "generic" calculation we could use in a module

Reply:
> 

Probably.